tests/kernel/iop/messagebox/send.c

```diff
@@ -0,0 +1,92 @@
+#include <common-iop.h>
+#include <thbase.h>
+#include <thmsgbx.h>
+#include "mbx-common.h"
+
+#define STACK_SIZE 0x800
+
+u32 g_receivedPayload = 0;
+
+void messageWaitThreadProc(u32 mbxId) {
+	MSG* msg = NULL;
+	ReceiveMbx((void**)&msg, mbxId);
+	g_receivedPayload = msg->payload;
+}
+
+void testMultipleSend(u32 mbxAttr) {
+	static const int msgCount = 3;
+	int i = 0;
+	MSG msg[msgCount];
+	
+	iop_mbx_t mbxInfo = {};
+	mbxInfo.option = 0;
+	mbxInfo.attr   = mbxAttr;
+	s32 mbxId = CreateMbx(&mbxInfo);
+
+	for(i = 0; i < msgCount; i++) {
+		memset(&msg[i], 0, sizeof(MSG));
+		msg[i].header.priority = msgCount - i;
+		msg[i].payload         = i + 1;
+		SendMbx(mbxId, &msg[i]);
```

Comment:
Does anything interesting happen if the `next` value already points to another message?  For example, can I send two messages at once?  I'm assuming it won't work, but I wonder if it will actually clear the `next` value to null or just leave it.  That might be important behavior.

-[Unknown]